addons/mail/wizard/mail_compose_message.py

```diff
@@ -328,6 +338,12 @@ def get_mail_values(self, res_ids):
                     attachment_ids,
                     {'model': 'mail.message', 'res_id': 0}
                 )
+                # Filter out the blacklisted records by setting the mail state to cancel -> Used for Mass Mailing stats
+                if blacklist:
```

Comment:
You may initialize blacklisted_rec_ids to a void list and remove this if: if res_id in blacklisted_rec_ids: ...